Tella/Data/Networking/Repositories/Dropbox/DropboxRepository.swift

```diff
@@ -0,0 +1,465 @@
+                if let error = error {
+                    if self.isDropboxAuthError(error) {
+                        continuation.resume(throwing: APIError.noToken)
+                    } else {
+                        continuation.resume(throwing: APIError.dropboxApiError(error))
+                    }
+                } else if let metadata = response?.metadata {
+                    continuation.resume(returning: metadata)
+                } else {
+                    continuation.resume(throwing: APIError.errorOccured)
+                }
+            }
+        }
+        
+        let folderId = metadata.id
+        let folderName = metadata.name
+        
+        // Upload the description file if it exists
+        try await uploadDescriptionFile(client: client, folderPath: "/\(folderName)", description: description)
+        
+        return DropboxCreateFolderResponse(id: folderId, name: folderName)
+    }
+    
+    private func uploadDescriptionFile(
+        client: DropboxClient,
+        folderPath: String,
+        description: String
+    ) async throws {
+        let descriptionData = description.data(using: .utf8) ?? Data()
+        
+        try await withCheckedThrowingContinuation { (continuation: CheckedContinuation<Void, Error>) in
+            client.files.upload(path: "\(folderPath)/description.txt", input: descriptionData)
+                .response { response, error in
+                    if let error = error {
+                        if self.isDropboxAuthError(error) {
+                            continuation.resume(throwing: APIError.noToken)
+                        } else {
+                            continuation.resume(throwing: APIError.dropboxApiError(error))
+                        }
+                    } else {
+                        continuation.resume(returning: ())
+                    }
+                }
+        }
+    }
+    
+    private func uploadFiles(
+        to folderPath: String,
+        files: [DropboxFileInfo]?,
+        subject: PassthroughSubject<DropboxUploadResponse, APIError>
+    ) async throws {
+        guard let files = files else {
+            throw APIError.errorOccured
+        }
+
+        try await ensureSignedIn()
+        guard let client = self.client else {
+            throw APIError.noToken
+        }
+
+        for (index, file) in files.enumerated() {
+            if isCancelled {
+                return
+            }
+
+            let fileState = FileUploadState(
+                fileURL: file.url,
+                fileName: file.fileName,
+                fileId: file.fileId,
+                sessionId: file.sessionId,
+                offset: file.offset ?? 0,
+                totalBytes: (try? FileManager.default.attributesOfItem(atPath: file.url.path)[.size] as? Int64) ?? 0
+            )
+            
+            do {
+                try await uploadFile(
```

Comment:
In my opinion Sending all the files in parallel is better than the serial upload to follow Tella web and Nextcloud implementation